library/core/functions.validation.php

```diff
@@ -244,14 +242,18 @@ function ValidateMinimumAge($Value, $Field, $FormPostedValues) {
 }
 
 if (!function_exists('ValidateInteger')) {
-   function ValidateInteger($Value, $Field = NULL) {
-      if (!$Value || (is_string($Value) && !trim($Value)))
-         return TRUE;
-
-      $Integer = intval($Value);
-      $String = strval($Integer);
-      return $String == $Value ? TRUE : FALSE;
-   }
+    function ValidateInteger($Value, $Field = null) {
+        if (!$Value || (is_string($Value) && !trim($Value))) {
+            return true;
+        }
+        $Integer = intval($Value);
+        $String = strval($Integer);
+        if ($String == $Value) {
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
You can probably remove this whole `if ($String == $Value)...` conditional.  It seems kind of redundant, based on the return values.  I think you can get away with just: `return $String == $Value;`